src/Compilers/CSharp/Portable/Symbols/Source/GlobalExpressionVariable.cs

```diff
@@ -43,14 +43,14 @@ internal class GlobalExpressionVariable : SourceMemberFieldSymbol
             Debug.Assert(nodeToBind.Kind() == SyntaxKind.VariableDeclarator || nodeToBind is ExpressionSyntax);
 
             var syntaxReference = syntax.GetReference();
-            return typeSyntax.IsVar
+            return (typeSyntax == null || typeSyntax.IsVar)
                 ? new InferrableGlobalExpressionVariable(containingType, modifiers, typeSyntax, name, syntaxReference, location, containingFieldOpt, nodeToBind)
                 : new GlobalExpressionVariable(containingType, modifiers, typeSyntax, name, syntaxReference, location);
         }
 
 
         protected override SyntaxList<AttributeListSyntax> AttributeDeclarationSyntaxList => default(SyntaxList<AttributeListSyntax>);
-        protected override TypeSyntax TypeSyntax => (TypeSyntax)_typeSyntax.GetSyntax();
+        protected override TypeSyntax TypeSyntax => (TypeSyntax)_typeSyntax?.GetSyntax();
```

Comment:
>TypeSyntax [](start = 38, length = 10)

Are all the consumers prepared to deal with null?

Reply:
Yes, so far as I have been able to determine.

---
In reply to: [156529635](https://github.com/dotnet/roslyn/pull/23209#discussion_r156529635) [](ancestors = 156529635)